tensorflow/python/util/tf_inspect.py

```diff
@@ -45,6 +45,27 @@ def getargspec(object):  # pylint: disable=redefined-builtin
                if d.decorator_argspec is not None), _inspect.getargspec(target))
 
 
+def getfullargspec(object):  # pylint: disable=redefined-builtin
+  """TFDecorator-aware replacement for inspect.getargspec.
+
+  Args:
+    object: A callable, possibly decorated.
+
+  Returns:
+    The `FullArgSpec` (Python 3) or `ArgSpec` (Python 2)
+    that describes the signature of the outermost decorator that
+    changes the callable's signature. If the callable is not decorated,
+    `inspect.getfullargspec()/inspect.getargspec()`
+    will be called directly on the callable.
+  """
+  # The getfullargspec is called in estimator.Estimator (See issue 12249)
+  decorators, target = tf_decorator.unwrap(object)
```

Comment:
If you unwrap it, it won't be the outermost object. I.e. if I have a decorator modifying the arguments, we should return the arguments directly, without unwrapping.